signer/tests/integration/postgres.rs

```diff
@@ -1381,6 +1389,106 @@ async fn fetching_deposit_request_votes() {
     signer::testing::storage::drop_db(store).await;
 }
 
+#[tokio::test]
+async fn fetching_deposit_signer_decisions() {
+    let pg_store = testing::storage::new_test_database().await;
+    let mut rng = rand::rngs::StdRng::seed_from_u64(51);
+
+    // This is just a sql test, where we use the `TestData` struct to help
+    // populate the database with test data. We set all the other
+    // unnecessary parameters to zero.
+    let num_signers = 3;
+    let test_model_params = testing::storage::model::Params {
+        num_bitcoin_blocks: 5,
+        num_stacks_blocks_per_bitcoin_block: 0,
+        num_deposit_requests_per_block: 1,
+        num_withdraw_requests_per_block: 0,
+        num_signers_per_request: num_signers,
+        concatenate_blocks: true,
+    };
+
+    let signer_set = testing::wsts::generate_signer_set_public_keys(&mut rng, num_signers);
+
+    let mut test_data = TestData::generate(&mut rng, &signer_set, &test_model_params);
+    test_data.write_to(&pg_store).await;
+
+    let signer_pub_key = signer_set.first().unwrap();
+
+    // move the oldest deposit request to the last block
+    // so that we change its created_at time to be the latest
+    test_data.deposit_requests.rotate_left(1);
+    let mut new_time_block = OffsetDateTime::now_utc() - time::Duration::minutes(15);
+    let mut new_time_request = OffsetDateTime::now_utc() - time::Duration::minutes(12);
+    // Update Bitcoin blocks
+    for (block, deposit) in test_data
```

Comment:
nit: can we split this into two different for loops? It doesn't seem we are doing anything with _both_ of them, and since we also changed the coupling above with the `rotate_left` is a bit confusing.